kilkari/src/main/java/org/motechproject/nms/kilkari/handler/SubscriptionManagerHandler.java

```diff
@@ -0,0 +1,85 @@
+package org.motechproject.nms.kilkari.handler;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.annotations.MotechListener;
+import org.motechproject.metrics.service.Timer;
+import org.motechproject.nms.kilkari.service.SubscriptionService;
+import org.motechproject.nms.kilkari.utils.KilkariConstants;
+import org.motechproject.scheduler.contract.CronSchedulableJob;
+import org.motechproject.scheduler.service.MotechSchedulerService;
+import org.motechproject.server.config.SettingsFacade;
+import org.quartz.CronExpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * Kikari subscription manager that takes care of purging subscriptions and maintaining current cap
+ */
+@Component
+public class SubscriptionManagerHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionManagerHandler.class);
+
+    @Autowired
+    @Qualifier("kilkariSettings")
+    private SettingsFacade settingsFacade;
+
+    @Autowired
+    private MotechSchedulerService schedulerService;
+
+    @Autowired
+    private SubscriptionService subscriptionService;
+
+    /**
+     * Use the MOTECH scheduler to setup a repeating job
+     * The job will start today at the time stored in flw.purge_invalid_flw_start_time in flw.properties
+     * It will repeat every flw.purge_invalid_flw_sec_interval seconds (default value is a day)
```

Comment:
This talks about a second interval, but the code below uses a cron expression. :confused: